plugins/saveactions.c

```diff
@@ -347,8 +706,13 @@ static gboolean on_document_focus_out(GObject *object, GeanyEditor *editor,
 PluginCallback plugin_callbacks[] =
 {
 	{ "document-new", (GCallback) &instantsave_document_new_cb, FALSE, NULL },
+	{ "document-new", (GCallback) &persistent_temp_files_document_new_cb, FALSE, NULL },
```

Comment:
I'd suggest having a single handler which then dispatches it to either the  instantsave case or the persistent temp file case. Same for `document-save` below.

Reply:
done